R/clusterSystems.R

```diff
@@ -0,0 +1,146 @@
+# ClusterSystems.R
+
+#' Cluster Systems
+#'
+#' \code{ClusterSystems} Input a list of n biological systems and re-cluster the genes into n clusters on an input variable of interest. Set overlap between the output clusters and clusters defined by input system membership is measured.
+#'
+#' The input systems are clustered according to the specified variables(s) of interest using PAM (partition around medoids) clustering. The similarity between input and output sets is measured using the Jaccard index of set overlap. P values for the observed Jaccard Indexes are calculated by measuring the Jaccard index of 1000 clusters randomly sampled from the output genes.
+#'
+#' @section Distance metrics :
+#' There are several different built in methods to compute the distance between a pair of genes.
+#'
+#' \code{"expression_profile"} uses GEO data to compute the similarity between 2 genes as the absolute value of the Spearmann correlation between their expression profiles. Distance is then taken as 1 - similarity.
+#'
+#' \code{"transcription_factor"} GTRD data to compute the number of shared transcription that bind upstream of the 2 genes divided by the total number of transcription factors. In effect, the Jaccard index of the sets of transcription factors from the 2 genes. Distance is then calculated as 1 - similarity.
+#'
+#' \code{"network_jaccard"} uses STRING network data to calculate the Jaccard similarity between the immediate neighbours of the 2 genes. Distance is then calculated as 1 - similarity.
+#'
+#' \code{"network_distance"} uses STRING network data and calculates the shortest path between between the 2 genes.
+#'
+#' @section More than one distance metric: If more than one distance data type is provided, the various distance metrics will be combined into one distance matrix according to one of the following methods:
+#'
+#' \code{'sum'} indicates the distances between a pair of genes will be defined as the sum of the distances according to each metric.
+#'
+#' \code{'product'} indicates that the distance between genes is defined as the product of their pairwise distances. This method more strongly penalizes genes that are distant by more than one metric.
+#'
+#' \code{'maximum'} or \code{'minimum'} indicate that the distance between 2 genes should be the maximum or minimum of the distances measured by each metric respectively.
+#'
+#' @section Custom distance metrics: In addition to several distance metrics which are built in to the function, the user has the option of defining a new distance metric to measure pairwise similarity between genes. If this is the case, an appropriate source of data (mapping from HGNC symbols to data of interest) that can be used as in input for each custom distance function.
```

Comment:
Add the parameter type (character, numeric ...), and if not obvious the dimension (vector, matrix) ... cf. the [review Request notes](https://github.com/hyginn/BCB420.2019.ESA/blob/master/PullRequestReviewNotes.md).